### ISSUE TITLE
QuoteFix for Mac Mojave 10.14

### ISSUE THREAD

**Issue by ykhemka** [robertklep/quotefixformac#89]
Hello,

Please advise when would the updates be available for macOS Mojave 10.14.

**Comment by robertklep** [comment]
To be honest: no idea.

Due to time constraints I haven't worked on Mojave-support at all, so once Mojave is released I will try and get it work ASAP. However, I've heard from other Mail plugin developers that it might not be trivial to get it working due to additional security measures imposed by Mojave.

**Comment by ykhemka** [comment]
Hello,

Thank you for your prompt reply.
I understand and hope that you continue with the new release.

Best regards,
Yash Khemka

Sent from my iPhone

On 23-Sep-2018, at 12:38 PM, Robert Klep <notifications@github.com> wrote:

To be honest: no idea.

Due to time constraints I haven't worked on Mojave-support at all, so once Mojave is released I will try and get it work ASAP. However, I've heard from other Mail plugin developers that it might not be trivial to get it working due to additional security measures imposed by Mojave.

**Comment by robertklep** [comment]
A part of what QuoteFix does, namely reformatting/cleaning up the message, could probably be implemented by means of a Mail rule that would match every new mail and run an (Apple)script on it, although I've never really tried that myself.

As for cursor placement (below the original message), that's really something that has to be done from Mail (or a plugin) itself.

**Comment by alexlevitskycom** [comment]
Thank you for your comment!

Maybe I'm wrong, but I have a feeling that the implementation of bottom posting using Apple Mail rules or Apple Automator scripting (which I have no idea how to implement), under quoted (original) message — is the only logical continuation of the tremendous QuoteFix & best decision in situation while QuoteFix doesn't work at Mojave (I hope that temporarily and the task will be solved with time).

**Comment by jklundell** [comment]
It's worth noting that a big part of QF's charm is that it piggybacks on the existing ⌘R and ⇧⌘R, along with their toolbar equivalents.  I can imagine a script of some sort that, once a top-posted reply was invoked, would select it, edit it, and replace it. And such a script might be better than doing it manually, but not nearly so compelling.

**Comment by jklundell** [comment]
This seems like a promising sign: https://gpgtools.tenderapp.com/kb/gpgmail-faq/enable-gpg-mail-on-macos-1014-mojave

**Comment by archodessa** [comment]
If anybody tried? QuoteFix works at Mojave now?

**Comment by robertklep** [comment]
If only it was that easy 😞 It's not just allowing the plugin to load or adding the correct UUID, the plugin crashes almost immediately because of changes made to Mail.

**Comment by jklundell** [comment]
Yeah. I'm _not_ suggesting that it's as simple a matter as enabling QuoteFix, just that Mail plugins are at least not ruled out in Mojave.

**Comment by dnblankedelman** [comment]
I know this is not even in the same league as my much mourned QuoteFix, but perhaps as a terrible stopgap: https://daringfireball.net/2007/07/non_top_posting_scripts . 

I have not been able to bring myself to try it, but wanted to report on something I found while looking to see if anyone else has done anything even remotely as amazing as Robert's work on his plugin.

**Comment by mgguinne** [comment]
@dnblankedelman I have been thinking there might be a way to create an AppleScript to do this instead of QF (will not be anywhere as good though), the script you gave is a good start but I wonder how to copy the email name and insert it? Maybe someone has already done this and I just should search myself...

**Comment by jimklo** [comment]
It seems like Apple now must have somewhat honest to goodness support for Plug-ins in Mojave, given there is now a "Manage Plug-ins..." Button and interface on the General Preferences tab of Mojave Mail.  MailHub recently added functional support, so I'm hoping there is at least a better path forward for the Mail plug-in society.

**Comment by defrilitus** [comment]
[Reply With Header](https://myjeeva.com/replywithheader) seems like it could be a stop-gap solution until QuoteFix can be updated. Certainly not as good, but better than nothing.

**Comment by glgray** [comment]
Which setting in RWH mimics QuoteFix's top-quoting behavior? I am not seeing it.

**Comment by defrilitus** [comment]
I used QuoteFix only for the extensive reply-header options. Sorry.

**Comment by gravesit** [comment]
I was able to get QuoteFix to work in Mojave by adding the following to Info.plist within the QuoteFix.mailbundle by right clicking on the bundle and "Show Package Contents":

```
@@ -104,6 +104,11 @@
        <array>
                <string>C86CD990-4660-4E36-8CDA-7454DEB2E199</string>
        </array>
+       <key>Supported10.14PluginCompatibilityUUIDs</key>
+       <array>
+               <string>A4343FAF-AE18-40D0-8A16-DFAE481AF9C1</string>
+               <string>A4343FAF-AE18-40D0-8A16-DFAE481AF9C1</string>
+       </array>
        <key>SupportedPluginCompatibilityUUIDs</key>
        <array>
                <string>225E0A48-2CDB-44A6-8D99-A9BB8AF6BA04</string>
@@ -155,6 +160,7 @@
                <string>DAF41AB7-F9AD-4273-9934-C81C74705B69</string>
                <string>1550C683-EA48-4036-B7CE-FB6F5D13EE02</string>
                <string>71562B89-0D90-4588-8E94-A75B701D6443</string>
+               <string>C86CD990-4660-4E36-8CDA-7454DEB2E199</string>
        </array>
 </dict>
```

Modified the install script from [ReplyWithHeader](https://myjeeva.com/replywithheader) to work for installing QuoteFix by changing "ReplyWithHeader" to "QuoteFix" and running the script from the `QuoteFix-v2.9.0` directory. Then all you have to do is enable in mail preferences -> General -> Manage Plug-ins... and restart mail.

**Comment by defrilitus** [comment]
Works perfectly! Thank you ever so much :-)

**Comment by dnblankedelman** [comment]
@gravesit That sounds very promising. 

Back in Oct, @robertklep said "the plugin crashes almost immediately because of changes made to Mail." Are you not seeing that behavior any more?

**Comment by robertklep** [comment]
Yeah, I'm interested too! 😄

**Comment by gravesit** [comment]
@dnblankedelman Been 24 hours with mail open and zero issues using the [install method outlined.](https://github.com/robertklep/quotefixformac/issues/89#issuecomment-438886331)

**Comment by defrilitus** [comment]
I installed the fix manually, but it's working fine all the same. Great news for lovers of this plug-in!

**Comment by sojuzpl** [comment]
it really works! Thx @gravesit

**Comment by ykhemka** [comment]
Hello,

I am unable to proceed, can you please send the info.plist file.

**Comment by defrilitus** [comment]
> Hello,
> 
> I am unable to proceed, can you please send the info.plist file.

Here is the whole bundle: QuoteFix 2.9 by @robertklep, including info.plist modifications by @gravesit. Many thanks to both of them 👍 

[QuoteFix.mailbundle.zip](https://github.com/robertklep/quotefixformac/files/2587640/QuoteFix.mailbundle.zip)

**Comment by ykhemka** [comment]
Thank you.

Best regards
Yash Khemka

**Comment by robertklep** [comment]
I just released [v2.10.0-alpha.1](https://github.com/robertklep/quotefixformac/releases/tag/v2.10.0-alpha.1).

I haven't heard back from the developer of ReplyWithHeader if it's okay to include their install scripts, so I went ahead and included them for the time being.

Please let me know (thumb-up this comment) if the installation process works okay, so I can do a proper stable release.

**Comment by alexlevitskycom** [comment]
Works perfect.

**Comment by dnblankedelman** [comment]
@robertklep related aside (since I tried this before your release, hence not being a direct thumbs up): I did a manual update of the plugin using the info the ReplyWithHeader shell script (ultimately just a cp command plus permission fixes). Before I could do that, I had to grant Terminal Full Disk Access. I do not know whether or not shell scripts individually need to be given that permission, but I thought I should bring it up just in case it is a thing.

The manual update to last stable with the new plist and the previous code has been working peachy for me.

**Comment by robertklep** [comment]
@dnblankedelman yeah, you need to grant Terminal those rights :) In my release, it's documented in the accompanying README file on how to do that.

**Comment by dnblankedelman** [comment]
Ah, you rock (as usual).

**Comment by jklundell** [comment]
a1 worked fine as upgrade, except the last message 'Follow the "Step 3" from "Install-Instructions-Mojave-OS-and-Above.txt' needs revision.

**Comment by robertklep** [comment]
@jklundell oops, cut&pasted a bit too much ;) thanks!

**Comment by alexlevitskycom** [comment]
Apple Mail restarts ofter after clicking on reply, with the last update for Mojave.
<img width="545" alt="screen shot 2018-12-01 at 00 23 20" src="https://user-images.githubusercontent.com/8191360/49317969-5eda4200-f4ff-11e8-9bda-c249975073d7.png">

And shows this message.
<img width="545" alt="screen shot 2018-12-01 at 00 17 32" src="https://user-images.githubusercontent.com/8191360/49317971-61d53280-f4ff-11e8-827d-6e1c6c67d131.png">

Any ideas how to avoid?

**Comment by robertklep** [comment]
@alexlevitskycom I assume that the plugin is causing Mail to crash. As for the reason, I have no idea without more information. You could fire up the Console app to try and find out if any reason is being logged what is causing the crash.

**Comment by mgguinne** [comment]
> @alexlevitskycom I assume that the plugin is causing Mail to crash. As for the reason, I have no idea without more information. You could fire up the Console app to try and find out if any reason is being logged what is causing the crash.

I have the same issue, sorry this was happening when you first released it and just did not worry (thought it was just me), though now someone else is having the same issue it would be nice to know how to fix.

However, I just went to send you some logs, I enabled it got the error message (forgot to get the crash log as mail reopened). I then went back in and enabled it again in mail, let it restart and then boom it works. Very strange.

**Comment by archodessa** [comment]
After mac resetting, the plugin is unchecked at Mail plugin manager, usually.

**Comment by ykhemka** [comment]
You would to check the same and also grant full access through privacy.

Best regards,
Yash Khemka

Sent from my iPhone

On 03-Dec-2018, at 4:57 PM, Architecture of Odessa <notifications@github.com> wrote:

After mac resetting, the plugin is unchecked at Mail plugin manager, usually.

**Comment by defrilitus** [comment]
@ykhemka Could you please explain what you mean by "grant full access through privacy"? If you mean QuoteFix, I see no way to do it (Security & Privacy settings in System Preferences are for apps only).
I am having the same problem as @archodessa: every time I close and restart Mail (or my Mac, I'm not sure which), the plugin is unchecked in Mail Plugin Manager.

**Comment by ykhemka** [comment]
Hello,

I am facing the same problem as of now and had to uninstall the QuoteFix.

Best regards
Yash Khemka

**Comment by robertklep** [comment]
@defrilitus I think they meant "grant full disk access to the Mail app".

Like I said, I had the issue of QF being disabled after a reboot, but it doesn't get disabled when restarting Mail (I did set the Full Disk Access after the reboot, and haven't rebooted since, so can't comment if that is the reason why it's working for me between restarts of Mail).

**Comment by defrilitus** [comment]
@robertklep QF is indeed being disabled after a reboot, but not after restarting the Mail app. For the moment, it seems that granting the Mail app FDA fixes the bug. I'll do some further testing this week to check if FDA is "the" solution here.

**Comment by defrilitus** [comment]
@robertklep After a couple of days of testing, it seems that granting the Mail app full disk access does prevent QuoteFix from being automatically disabled. The only exception, of course, is when macOS (and thus the Mail app) is updated to a new version, but that is to be expected. Maybe a note suggesting enabling FDA for the Mail app should be in the installation instructions?

**Comment by mgguinne** [comment]
I got this to work, I was having issues with the preferences not coming up once added (mail hanging). Once I gave Mail app full disk access, no further issues. 

Thanks for this @robertklep , it's really appreciated.
